audit-checklists/Arbitrum.md

```diff
@@ -0,0 +1,95 @@
+
+### Permission to cancel the retryable ticket
+
+When creating new retryable ticket one of the parameters that is provided to the `Inbox::createRetryableTicket` is `callValueRefundAddress`. This is the address where `l2CallValue` is refunded to in the case that retryable ticket execution fails on L2. But additionally `callValueRefundAddress` has the permission to cancel the ticket, making it permanently unclaimable. It is important to keep this in mind in case protocol is designed in a way where this can be exploited. Here's an example of such scenario:
+
+- protocol has permissionless function on L1 that lets anyone create retryable ticket to execute action X on the L2
+- protocol assumes once retryable ticket is submitted action X will surely be executed on L2
+  - there is no way to re-trigger the action X
+- attacker calls the L1 function to submit retryable ticket but intentionally provides too small gas limit to make sure ticket can't be auto-redeemed on L2
+- attacker also provides own address for `callValueRefundAddress`
+- retryable ticket auto-redemption fails and attacker cancels the ticket before anyone else can redeem it
+- as a consequence, action X can never get executed
+
+> Make sure the account provided as `callValueRefundAddress` cannot misuse its capability to cancel the ticket in a way that hurts the protocol
+
+Bug examples: [1](https://github.com/code-423n4/2024-05-olas-findings/issues/29)
+
+## Oracle integration
+
+### Sequencer uptime
+
+Typically users interact with Arbitrum through sequencer. Sequencer is responsible for including and processing transactions. If sequencer goes down users can still submit transaction using [Inbox](https://etherscan.io/address/0x4Dbd4fc535Ac27206064B68FfCf827b0A60BAB3f) contract on L1. However, such transactions will be included with potentially significant delay. In the worst case, if sequencer is down for a prolonged time, transactions submitted to L1 can be force included only after 24h. In the context of price feeds, it means that reported prices can get stale and significantally drift from the actual price. Thus protocols should always check the sequencer's uptime and revert the price request if sequencer is down for more than predefined period of time.
+
+Chainlink's sequencer uptime feeds can be used for this purpose. Integration example can be found [here](https://docs.chain.link/data-feeds/l2-sequencer-feeds#example-code).
+
+> Integrations with Chainlink's price feeds on Arbitrum should always check the sequencer uptime in order to avoid stale prices
+
+Bug examples: [1](https://code4rena.com/reports/2024-07-benddao#m-12--no-check-if-arbitrumoptimism-l2-sequencer-is-down-in-chainlink-feeds-priceoraclesol-), [2](https://github.com/shieldify-security/audits-portfolio-md/blob/main/Pear-v2-Security-Review.md#m-02-missing-check-for-active-l2-sequencer-in-calculatearbamount)
```

Comment:
Add this example: https://github.com/sherlock-audit/2023-05-perennial-judging/issues/37